### PR TITLE
 Vampire, antag datums : take 2 

### PR DESCRIPTION
**Pull request by ShiftyRail**
Doing a quick PR before I go to bed
Done : 
- [x] Stone mask
- [x] Refactoring vampire powers to something modular and fancy

To note : 
- Changed `protected_jobs` into `restricted_jobs` in `CanBeAssigned()`. A _restricted_ job cannot be that role, ever, think a Chaplain becoming a vampire. A _protected_ job can be that role, but with a reduced chance, like security officers rolling antags.

To do (in a later PR) :
- [ ] Like a dozen spells or so
- [ ] Enthralling and de-enthralling (hopefully this should be easy with the role system)

### CODE REVIEW

**Comment by gbasood** [comment]
>gee sood

listen do you think I actually know how role datums are supposed to work
because I don't
I just showed up to fix the bad merge

**Comment by ShiftyRail** [comment]
@gbasood it's just harmless banter, no worries
You did a good job fixing it

File: code/datums/gamemode/role/role.dm [hunk 1]
```diff
@@ -124,7 +124,10 @@
 	M.antag_roles.Add(id)
 	M.antag_roles[id] = src
 
-	OnPreSetup()
+	if (!OnPreSetup())
+		return FALSE
+	if (!OnPostSetup())
```
Comment (MadmanMartian) [inline code/datums/gamemode/role/role.dm]: Please do not call OnPostSetup here. It's to be called after the game has been set up, and AssignToRole's called before that.

**Review by MadmanMartian — CHANGES_REQUESTED** [review]

**Comment by ShiftyRail** [comment]
>Conflicts

Wew lad

**Comment by DeityLink** [comment]
Vampires and Cultists have similar weaknesses, sorry man

File: vgstation13.dme [hunk 2]
```diff
@@ -2165,6 +2166,7 @@
 #include "code\modules\spells\general\rune_write.dm"
 #include "code\modules\spells\targeted\balefulmutate.dm"
 #include "code\modules\spells\targeted\buttbots_revenge.dm"
+#include "code\modules\spells\targeted\disaese.dm"
```
Comment (gbasood) [inline vgstation13.dme]: >disaese.dm

File: code/datums/gamemode/vampire_gamemode.dm [hunk 3]
```diff
@@ -1,4 +1,122 @@
...
+	id = VAMP_SHAPE
+	spell_path = /spell/shapeshift
+
+/datum/power/vampire/vision
+	blood_threeshold = 100
+	id = VAMP_VISION
+	spell_path = null // TODO
+
+/* Tier 2 */
+/datum/power/vampire/disaese
```
Comment (ghost) [inline code/datums/gamemode/vampire_gamemode.dm]: >disaese

je pense c'est peut-etre une erreur

Comment (ShiftyRail) [reply code/datums/gamemode/vampire_gamemode.dm]: There he goes again. Look, everyone! He posted it once again! Isn't he just the funniest guy around?!

File: code/datums/gamemode/role/role.dm [hunk 4]
```diff
@@ -156,8 +157,8 @@
 // General sanity checks before assigning antag.
 // Return 1 on success, 0 on failure.
 /datum/role/proc/CanBeAssigned(var/datum/mind/M)
-	if(protected_jobs.len>0)
-		if(M.assigned_role in protected_jobs)
+	if(restricted_jobs.len>0)
```
Comment (MadmanMartian) [inline code/datums/gamemode/role/role.dm:160]: I feel this has changed multiple times

**Comment by ShiftyRail** [comment]
Oh my it compiled just fine on my machine
Will fix it

**Review by MadmanMartian — APPROVED** [review]

File: code/datums/gamemode/role/role.dm (hunk 4 above)
Comment (ShiftyRail) [reply code/datums/gamemode/role/role.dm:160]: Sorry lad

**Pull request merged**